pkg/acpi/bios.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package acpi
+
+import "fmt"
+
+const (
+	llDSDTAddr = 140
+	lDSDTAddr  = 40
+)
+
+// BiosTable contains the information needed to create table images for
+// firmware such as coreboot or oreboot. It hence includes the RSDP,
+// [XR]SDT, and the raw table data. The *SDT is always Tables[0]
+// as in the real tables.
+type BiosTable struct {
+	RSDP   *RSDP
+	Tables []Table
+}
+
+// ReadBiosTables reads tables that are not interpreted by the OS,
+// i.e. it goes straight to memory and gets them there. We optimistically
+// hope the bios has not stomped around in low memory messing around.
+func ReadBiosTables() (*BiosTable, error) {
+	r, err := GetRSDPEBDA()
+	if err != nil {
+		r, err = GetRSDPMem()
+		if err != nil {
+			return nil, err
+		}
+	}
+	Debug("Found an RSDP at %#x", r.base)
+	x, err := NewSDTAddr(r.SDTAddr())
+	if err != nil {
+		return nil, err
+	}
+	Debug("Found an SDT: %s", String(x))
+	bios := &BiosTable{
+		RSDP:   r,
+		Tables: []Table{x},
+	}
+	for _, a := range x.Addrs {
+		Debug("Check Table at %#x", a)
```

Comment:
I'd add more context to your debug prints, like "Decoding ACPI Table at phys %#x", or remove them

Reply:
I'm inclined to remove them at this point, it all seems to work.

Reply:
removed.